### PR TITLE
EWB-1554:  Add GRPC compression interceptor

### PR DESCRIPTION
**Pull request by Lax125**

### CODE REVIEW

**Comment by ryanjcoleman** [comment]
Task linked: [EWB-1554 Add compression interceptor to evolve-conn-jvm](https://app.clickup.com/t/6929263/EWB-1554)

**Comment by Lax125** [comment]
If there's a need to generalize this (e.g. specify type of compression), let me know

**Comment by kgreav** [comment]
You should make the compression type at least be able to be passed in, and have it default to gzip
i'm not sure what other compression types are possible, but it's obviously configurable so we should allow for that.

File: src/main/kotlin/com/zepben/evolve/conn/grpc/CompressionInterceptor.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import io.grpc.ServerInterceptor
+
+class CompressionInterceptor : ServerInterceptor {
+
+    override fun <ReqT : Any?, RespT : Any?> interceptCall(
+        call: ServerCall<ReqT, RespT>?,
+        headers: Metadata?,
+        next: ServerCallHandler<ReqT, RespT>?
+    ): ServerCall.Listener<ReqT>? {
+        call?.setCompression("gzip")
```
Comment (kgreav) [inline src/main/kotlin/com/zepben/evolve/conn/grpc/CompressionInterceptor.kt:23]: make the `"gzip"` be passed in to the CompressionInterceptor

**Review by kgreav — CHANGES_REQUESTED** [review]

**Review by kgreav — APPROVED** [review]

**Pull request merged**